test/WebJobs.Script.Tests/Security/BlobStorageSecretManagerTests.cs

```diff
@@ -0,0 +1,595 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Azure.WebJobs.Script.WebHost.Properties;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Moq;
+using Newtonsoft.Json;
+using WebJobs.Script.Tests;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Security
+{
+    public class BlobStorageSecretManagerTests : IClassFixture<BlobStorageSecretManagerTests.Fixture>
+    {
+        private readonly BlobStorageSecretManagerTests.Fixture _fixture;
+        private ScriptSettingsManager _settingsManager = ScriptSettingsManager.Instance;
+
+        public BlobStorageSecretManagerTests(BlobStorageSecretManagerTests.Fixture fixture)
+        {
+            //TODO: merge this into SecretManagerTests with SecretRepository options so that they run the same tests for both File and Blob repositories without as much duplication
```

Comment:
Yes, we'd need to merge if there are additional tests we'd like to run going through the `SecretManager`, a lot (most) of the tests in this file though are just testing the behavior of the secret manager itself, so I'm not sure we need to duplicate the tests using the blob repository. The file repository was only used here because that was the initial implementation (when none of those abstractions were in place), but we could probably just update these tests to go against a mock and not a concrete repository implementation. 
Just want to make sure we're not increasing the test maintenance if the surface tested is the same. I'm assuming the blob repository is tested in the other test class you've added for it.